### ISSUE TITLE
create a multilingual translation project for Nim language documents

### ISSUE THREAD

**Issue by VectorWorkshopBaoErJie** [nim-lang/Nim#18821]
### Summary
It is recommended to create a multilingual translation project for NIM language documents.  
### Description
It is recommended to create a multilingual translation project for NIM language documents.  Thus, I am willing to complete the Chinese translation of the document.  For example, check out https://weblate.org/, where I've been involved in translating a lot of Godot documents.  This has enabled Godot to maintain a good momentum of user growth in China.

**Comment by xflywind** [comment]
PR is welcome

**Comment by VectorWorkshopBaoErJie** [comment]
Given the need to translate into multiple languages, it may be necessary to create a separate library.

**Comment by ringabout** [comment]
There are two parts of Nim docs:
- standard library docs
- manual, tutorial etc. (https://github.com/nim-lang/Nim/tree/devel/doc)

IMO The second one can be translated first as a start. I would like a switch button to shift languages.

For instance, the Python documentation handles it well

![image](https://user-images.githubusercontent.com/43030857/132444722-98f587ff-6aa9-4ae6-9aef-d15c26be1a46.png)

**Comment by ghost** [comment]
If this gets implemented, I can start contributing Russian translations. Also, some things like the tutorial were already translated to Russian by someone else in ~2015, so it's pretty easy to bring them up-to-date

**Comment by ringabout** [comment]
Hi @qjyqjyqjyqjy 

>  it may be necessary to create a separate library.

Feel free to create a separate library which can be moved into `nim-lang` org later (if needed).

**Comment by VectorWorkshopBaoErJie** [comment]
https://github.com/nim-lang/website/tree/master/jekyll 
This page doesn't seem to have a good workflow to keep updating the translation to add content.

**Comment by juancarlospaco** [comment]
Would be better to implement it in the docgen somehow, instead of special case it.

**Comment by FedericoCeratto** [comment]
Related RFC: https://github.com/nim-lang/RFCs/issues/96
Previous PR: https://github.com/nim-lang/Nim/pull/8842

**Comment by ringabout** [comment]
@FedericoCeratto Hello, is it possible to set up a https://weblate.org/ without nimdoc changes? Since manuals and tutorials are written by rst files.

**Comment by juancarlospaco** [comment]
Weblate would be great, anyone can translate from browser.

**Comment by FedericoCeratto** [comment]
@ringabout I proposed using weblate in https://github.com/nim-lang/RFCs/issues/96.
Yet, the manual and the library documentation are generated by Nim. At some point nimdoc has to pull in the translation files from weblate to output the HTML doc.

As a side note, having translation support in nimdoc would also enable Nim developers to translate the documentation of their own projects.

**Comment by ringabout** [comment]
Yeah, I'm in favor of using `weblate`. In a progressive view, I would like to have a `weblate` process first. I'm not familiar with its workflow. I would like to see a template so that I can set up a process for some Nim documentation (I tried but failed before).

Or we can maintain a forked nim doc extension to get the work done using the compilerapi, somehow. Now that we have `std/importutils` which make it easier to use internal procs and data structures. It seems feasible to me. Then we can set up the translation in a separate repo, pulling the Nim source from the main repo daily. @FedericoCeratto I can try to implement https://github.com/nim-lang/Nim/pull/8842 as a tool this weekend if I have some free time.
